auth_backend/backends.py

```diff
@@ -34,10 +34,25 @@ def authenticate(self, email=None, username=None, password=None, **kwargs):
         if strategy:
             payload['strategy'] = strategy
 
+        auth_api_client = AuthApiClient(cas_credentials)
         status, data = auth_api_client.call('sessions', 'POST', payload)
 
         if status not in (200, 404,):
             raise CASUnexpectedStatusCode(status, data)
 
         if status == 200:
-            return user
+            if existing_user:
+                existing_user.cas_credentials = cas_credentials
+            else:
+                try:
+                    # Do not on save sync to CAS, as we just got the user's
+                    # data from CAS, and nothing has changed in the interim
+                    pre_save.disconnect(save_user_to_cas, sender=KagisoUser)
+                    existing_user = KagisoUser(cas_credentials)
+                    existing_user.set_password(password)
+                    existing_user.create_from_cas_data(data)
+                    existing_user.save()
+                finally:
+                    pre_save.connect(save_user_to_cas, sender=KagisoUser)
```

Comment:
Why did you switch from using a `Manager` to using the `hooks`

i dont like how you switch off the hook, when you dont want to sync, it does not seem linear ...

cant you rather do a `is__not_new` check in your `save_user_to_cas` function, that way it wont sync if the user instance has just been created